### PR TITLE
Fix sip wcs handling

### LINKED ISSUE #457: Try harder to correctly handle WCS with SIP distortion
The way that SIP distortions in a WCS are handled changed in astropy 1.2 when converting a WCS to a header. See https://github.com/astropy/astropy/pull/4814 for details.

In particular, if a WCS has been created with SIP distortion coefficients *and* the `CTYPE` contains `SIP` then the `SIP` is dropped from the CTYPE when converting the WCS back to a header using `to_header` *unless* `relax=True` in the call to `to_header`.  See extensive discussion at the PR above and in the issues linked to from the PR for motivation.

Suppose you have an image the SIP distortion in the WCS, then do these steps:

```python
# Read the image...
img = CCDData.read('m101-007b.fit')
img.write('i-have-no-sip.fit')
```

you will get this warning:
```
WARNING: Some non-standard WCS keywords were excluded: BP_0_0, BP_0_1, BP_0_2, B_1_1, B_2_0, A_ORDER, B_0_2, B_ORDER, BP_ORDER, AP_ORDER, A_1_1, BP_2_0, AP_2_0, A_0_2, BP_1_1, BP_1_0, A_2_0, AP_1_0, AP_1_1, AP_0_1, AP_0_0, AP_0_2 Use the ``relax`` kwarg to control this. [astropy.wcs.wcs]
```

I think the desired outcome is to keep the distortion keywords if `SIP` is in `CTYPE`...i.e. the simple fix is to use `relax=True` in `to_header`.

In fact, the final image will, when read back in, will still apply SIP distortions because the distortions are never removed from the header (when we convert the WCS `to_header` we do not delete any WCS-related keywords already in the header).

### PR DESCRIPTION
**Pull request by mwcraig**
This fixes #457 by adding the relax keyword to `CCDData.to_header`. Without this change, files with `-SIP` as part of `CTYPE` have that truncated when converted from a `CCDData` to a header.

For bugfixes:

- [x] Did you add an entry to the "Changes.rst" file?
- [x] Did you add a regression test?
- [x] Does the commit message include a "Fixes #issue_number" (replace "issue_number").

For new functionality:

- [x] Did you add an entry to the "Changes.rst" file?
- [x] Did you include a meaningful docstring with Parameters, Returns and Examples?
- [x] Does the commit message include a "Fixes #issue_number" (replace "issue_number").
- [x] Did you include tests for the new functionality?

### CODE REVIEW

File: ccdproc/ccddata.py [hunk 1]
```diff
@@ -308,6 +308,12 @@ def to_hdu(self, hdu_mask='MASK', hdu_uncertainty='UNCERT',
             Default is ``'MASK'`` for mask, ``'UNCERT'`` for uncertainty and
             ``None`` for flags.
 
+        wcs_relax : bool
```
Comment (MSeifert04) [inline ccdproc/ccddata.py]: I think the description should also link to http://docs.astropy.org/en/stable/api/astropy.wcs.WCS.html#astropy.wcs.WCS.to_header.

**Review by MSeifert04 — APPROVED** [review]
LGTM.

**Comment by mwcraig** [comment]
Forgot that some of behavior here is astropy version dependent, fixing...

**Comment by mwcraig** [comment]
Merging...

**Pull request merged**